index.d.ts

```diff
@@ -1,5 +1,11 @@
 // TypeScript Version: 4.1
-type AxiosHeaders = Record<string, string | string[] | number | boolean>;
+type JSONValue = string | string[] | number | boolean;
+
+interface JSONObject {
```

Comment:
I think we could probably specify the potential keys more concretely. If I understand correctly, just `HTTP` methds in lower case, or `common` are allowed. So it could be something like
```ts
type HTTPMethod = 'get' | 'head' | 'post' | 'put' | 'delete' | 'connect' | 'options' | 'trace' | 'patch';
type HeaderValue = string | string[] | number | boolean;

type type AxiosHeaders = Record<string, HeaderValue | Record<HTTPMethod | 'common', HeaderValue>>;
```

Reply:
Great observation, but HTTPMethod has already been declared on https://github.com/axios/axios/blob/b4bd505828f409085d1fc341a1e1169889fb0ede/index.d.ts#L51-L61.

so I can replace this code https://github.com/axios/axios/blob/b4bd505828f409085d1fc341a1e1169889fb0ede/index.d.ts#L1-L6

to
```js
// TypeScript Version: 4.1
type HeaderValue = string | string[] | number | boolean;

type AxiosHeaders = Record<string, HeaderValue | Record<Method & CommonHeaders, HeaderValue>>;
```